### ISSUE TITLE
Installation fails due to conflicting openpyxl version

### ISSUE THREAD

**Issue by NeolithEra** [Thimxx/pyGenealogical-Tools#115]
Hi, users are unable to run **_pygenealogicaltools_** due to dependency conflict with _**openpyxl**_ package. As shown in the following full dependency graph of **_pygenealogicaltools_**, **_pyexcel-xlsx_** requires **_openpyxl <2.6.0,>=2.5.0_**，while **_pygenealogicaltools_** requires **_ openpyxl *_**.

According to pip’s _“first found wins”_ installation strategy, _**openpyxl 3.0.2**_ is the actually installed version. However, _**openpyxl 3.0.2**_ does not satisfy _**<2.6.0,>=2.5.0**_.

### Dependency tree-----------
```
pygenealogicaltools - 0.4.0
| +- langdetect(install version:1.0.7 version range:*)
| +- mapbox(install version:0.18.0 version range:*)
| | +- boto3(install version:1.10.39 version range:>=1.4)
| | | +- botocore(install version:1.13.39 version range:>=1.13.39,<1.14.0)
| | | | +-python-dateutil(install version:2.8.0 version range:<2.8.1,>=2.1）
| | | +- jmespath(install version:0.7.0 version range:>=0.7.1,<1.0.0)
| | | +- s3transfer(install version:0.2.1 version range:>=0.2.0,<0.3.0)
| | +- cachecontrol(install version:0.12.5 version range:*)
| | | +-msgpack(install version:0.6.2 version range:*)
| | +- iso3166(install version:1.0.1 version range:*)
| | +- polyline(install version:1.4.0 version range:>=1.3.1)
| | | +- six(install version:1.13.0 version range:>=1.8.0)
| | +- python-dateutil(install version:2.8.1 version range:>=2.5.0)
| | +- requests(install version:2.22.0 version range:*)
| | | +- certifi(install version:2019.11.28 version range:>=2017.4.17)
| | | +- chardet(install version:3.0.4 version range:<3.1.0,>=3.0.2)
| | | +- idna(install version:2.8 version range:>=2.5,<2.9)
| | | +- urllib3(install version:1.25.7 version range:<1.26,>=1.21.1)
| | +- uritemplate(install version:3.0.0 version range:>=2.0)
| +- metaphone(install version:0.6 version range:*)
| +- geotext(install version:0.4.0 version range:*)
| +- openpyxl(install version:3.0.2 version range:*)
| +- pyexcel(install version:0.5.15 version range:*)
| | +- lml(install version:0.0.9 version range:>=0.0.4)
| | +- pyexcel-io(install version:0.5.20 version range:>=0.5.19)
| | | +- lml(install version:0.0.9 version range:>=0.0.4)
| +- pyexcel-xls(install version:0.5.8 version range:*)
| | +- pyexcel-io(install version:0.5.20 version range:>=0.5.3)
| | | +- lml(install version:0.0.9 version range:>=0.0.4)
| | +- xlrd(install version:1.2.0 version range:*)
| | +- xlwt(install version:1.3.0 version range:*)
| | | +- coveralls(install version:1.9.2 version range:*)
| | | +- nose(install version:1.3.7 version range:*)
| | | +- nose-cov(install version:1.6 version range:*)
| | | +- nose-fixes(install version:1.3 version range:*)
| | | | +- nose(install version:1.3.7 version range:*)
| | | | +- setuptools(install version:42.0.2 version range:*)
| | | +- panci(install version:0.0.2 version range:*)
| | | | +- pyyaml(install version:5.2 version range:*)
| | | +- pkginfo(install version:1.5.0.1 version range:*)
| | | +- setuptools-git(install version:1.2 version range:*)
| | | +- six(install version:1.13.0 version range:*)
| | | +- sphinx(install version:2.2.2 version range:*)
| | | +- tox(install version:3.14.2 version range:*)
| | | +- twine(install version:3.1.1 version range:*)
| | | +- wheel(install version:0.33.6 version range:*)
| +- pyexcel-xlsx(install version:0.5.7 version range:*)
| | +- openpyxl(install version:2.5.14 version range:>=2.5.0,<2.6.0)
| | | +- et-xmlfile(install version:1.0.1 version range:*)
| | | | +- python(install version:2.5 version range:>=2.6.0)
| | | +- jdcal(install version:1.4.1 version range:*)
| | +- pyexcel-io(install version:0.5.20 version range:>=0.5.3)
| | | +- lml(install version:0.0.9 version range:>=0.0.4)
| +- python-levenshtein(install version:0.12.0 version range:*)
| | +- setuptools(install version:42.0.2 version range:*)
| +- requests(install version:2.22.0 version range:*)
| | +- certifi(install version:2019.11.28 version range:>=2017.4.17)
| | +- chardet(install version:3.0.4 version range:<3.1.0,>=3.0.2)
| | +- idna(install version:2.8 version range:>=2.5,<2.9)
| | +- urllib3(install version:1.25.7 version range:<1.26,>=1.21.1)
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Suggested Solution

1.	Fix his direct dependency to be **_openpyxl <2.6.0,>=2.5.0_**. I have checked this revision will not affect your downstream projects now.
2.	Ask your upstream project **_pyexcel-xlsx_** to loose the version range of **_openpyxl_**.
3.	Remove your direct dependency **_openpyxl_**, and use **_openpyxl_** transitively introduced by **_pyexcel-xlsx_**.

@Thimxx Which solution do you prefer, 1, 2 or 3?
Please let me know your choice. I can submit a PR to solve this issue.

**Comment by Thimxx** [comment]
Interesting finding, thanks. Actually I think I will go for option 3, it is more sensible one.
What's your urgency?

**Comment by Thimxx** [comment]
I will try to wait until update, as the project looks active, created the following issue: https://github.com/pyexcel/pyexcel-xlsx/issues/34

I will go for option 1 meanwhile

**Comment by Thimxx** [comment]
Fixed in version 0.4.2, I introduced the dependencies, but in parallel I am trying to work to free the dependencies in the other packages... I would prefer to have more flexibility. So I keep the issue open until that is fixed. Now is accessible through pip

**Comment by Thimxx** [comment]
The issue has been solved by the developer of pyexcel-xlsx, in the new version 0.4.3 the versions of pyexcel-xlsx shall be over or equal 0.5.8 and openpyxl over or equal 2.6.1. I consider this closed so, as not conflict of dependencies is in place and the trick done in 0.4.2 is now recovered back.
